### DISCUSSION TITLE
Trial level node is missing the "type" parameter.

### DISCUSSION THREAD

**Discussion by JulietDunstone** [jspsych/jsPsych#1390] [Q&A] [answered]
I have a section of code that used to be part of a huge function but I have split into smaller sections to make it more manageable and to allow more communication with a websocket server, and now that this section is a single function it no longer works. The only part that is new is the jsPsych.init and the fact that this is now a standalone function.

I get the following error:

"Trial level node is missing the "type" parameter. The parameters for the node are: {"0":{"type":"instructions","pages":["long list of instructions that I have removed for brevity!"],"show_clickable_nav":true},"1":{"type":"html-button-response","stimulus":"Press START to begin your training","choices":["START"]}}"

I have specified type for both nodes and I don't know where the 0 and 1 come from in the list of parameters given by the error.

full code:
```
function training_instructions(){
		var timeline=[];
		var instructions={
			type: 'instructions',
			pages: ['instr 1.',
			'instr2',
			'instr3',
			etc'],
			show_clickable_nav: true
		};
		var start_button={
			type: 'html-button-response',
			stimulus: 'Press START to begin your training',
			choices: ['START']
		};
		timeline.push(instructions, start_button);

		jsPsych.init({
	      timeline: [timeline],
	      on_finish: function(){
	        time = new Date().toLocaleTimeString([], {hour: '2-digit', minute:'2-digit',second:'2-digit', hour12: false});
					writeData({bla bla bla}),
	        ws.send(JSON.stringify({bla bla bla}))
	      }
	    })
}

**Comment by becky-gilbert** [comment, marked as answer]
Hi @JulietDunstone, this "Trial level node is missing the "type" parameter" error often happens when there's something wrong with the timeline. In your case, I think the problem is that the `timeline` variable is an array, but then you put it inside of another array in `jsPsych.init`, which turns it into a nested array. So you can just remove the square brackets around your timeline array:
```javascript
jsPsych.init({
    timeline: timeline,
    ...
```
Does that work?

**Reply by JulietDunstone** [reply]
yes! I clearly copied my jsPsych.init from a function that only had one thing in the timeline so it wasn't an array. Thank you so much!
